kbase-extension/static/kbase/js/common/ui.js

```diff
@@ -278,55 +295,15 @@ define([
         return window.confirm(prompt);
     }
 
-    function renderConfirmDialog(arg) {
```

Comment:
Refactored these dialog functions to remove a whole lot of duplicated code (ugh!). The click handlers were dodgy so I redid them as well, and added tests for everything.